hack/generator/pkg/codegen/pipeline_create_storage_types.go

```diff
@@ -7,247 +7,67 @@ package codegen
 
 import (
 	"context"
-	"fmt"
-
 	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/codegen/storage"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
 )
 
 // createStorageTypes returns a pipeline stage that creates dedicated storage types for each resource and nested object.
 // Storage versions are created for *all* API versions to allow users of older versions of the operator to easily
 // upgrade. This is of course a bit odd for the first release, but defining the approach from day one is useful.
-func createStorageTypes() PipelineStage {
+func createStorageTypes(idFactory astmodel.IdentifierFactory) PipelineStage {
 	return MakePipelineStage(
 		"createStorage",
 		"Create storage versions of CRD types",
 		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
 
-			storageTypes := make(astmodel.Types)
-			visitor := makeStorageTypesVisitor(types)
-			vc := makeStorageTypesVisitorContext()
-			var errs []error
-			for _, d := range types {
-				d := d
+			// Create a factory for each group (aka service) and divvy up the types
+			factories := make(map[string]*storage.StorageTypeFactory)
+			for name, def := range types {
 
-				if astmodel.ARMFlag.IsOn(d.Type()) {
-					// Skip ARM definitions, we don't need to create storage variants of those
+				ref, ok := name.PackageReference.AsLocalPackage()
+				if !ok {
+					// Skip definitions from non-local packages
+					// (should never happen)
```

Comment:
Consider logging a warning in this case at the least (assuming it really should never happen?)

Reply:
Added warning